pkg/internal/controllers/certificaterequests.go

```diff
@@ -122,19 +121,19 @@ func addCertificateRequestController(ctx context.Context, opts Options) error {
 		// Watch CertificateRequestPolicies. If a policy is created or updated,
 		// then we need to process all CertificateRequests that do not yet have an
 		// approved or denied condition since they may be relevant for the policy.
-		Watches(&source.Kind{Type: new(policyapi.CertificateRequestPolicy)}, handler.EnqueueRequestsFromMapFunc(enqueueRequestFromMapFunc)).
+		Watches(new(policyapi.CertificateRequestPolicy), handler.EnqueueRequestsFromMapFunc(enqueueRequestFromMapFunc)).
 
 		// Watch Roles, RoleBindings, ClusterRoles, and ClusterRoleBindings. If
 		// RBAC changes in the cluster then CertificateRequestPolicies may become
 		// appropriate for a CertificateRequest. On RBAC events, Reconcile all
 		// CertificateRequests that are neither Approved or Denied.
 		// Only need to cache metadata for RBAC resources since we do not need any
 		// information in the spec.
-		Watches(&source.Kind{Type: new(rbacv1.Role)}, handler.EnqueueRequestsFromMapFunc(enqueueRequestFromMapFunc), builder.OnlyMetadata).
-		Watches(&source.Kind{Type: new(rbacv1.RoleBinding)}, handler.EnqueueRequestsFromMapFunc(enqueueRequestFromMapFunc), builder.OnlyMetadata).
-		Watches(&source.Kind{Type: new(rbacv1.ClusterRole)}, handler.EnqueueRequestsFromMapFunc(enqueueRequestFromMapFunc), builder.OnlyMetadata).
-		Watches(&source.Kind{Type: new(rbacv1.ClusterRoleBinding)}, handler.EnqueueRequestsFromMapFunc(enqueueRequestFromMapFunc), builder.OnlyMetadata).
-		Watches(&source.Kind{Type: new(corev1.Namespace)}, handler.EnqueueRequestsFromMapFunc(enqueueRequestFromMapFunc), builder.OnlyMetadata).
+		WatchesMetadata(new(rbacv1.Role), handler.EnqueueRequestsFromMapFunc(enqueueRequestFromMapFunc)).
```

Comment:
This new watch type appears to be a wrapper that automatically appends the `OnlyMetadata` option https://github.com/kubernetes-sigs/controller-runtime/pull/2135/commits/dc62ed5dadc6ea2d3e22546368b939a2db1edfb2#diff-e77b9468ab935d5ea4d5cdae3b994114bada17df0001570e4b1436419afd50ccR166-R172